maintenance/projects/js-toolkit/packages/liferay-npm-build-support/src/scripts/start.js

```diff
@@ -4,26 +4,128 @@
  */
 
 import childProcess from 'child_process';
+import fs from 'fs';
+import FilePath from 'liferay-npm-build-tools-common/lib/file-path';
+import {
+	error,
+	print,
+	success,
+	warn,
+} from 'liferay-npm-build-tools-common/lib/format';
 import project from 'liferay-npm-build-tools-common/lib/project';
 import os from 'os';
 import path from 'path';
+import readJsonSync from 'read-json-sync';
 import util from 'util';
 
 import * as cfg from '../config';
 import {Renderer} from '../util';
 
 const templatesDir = path.join(__dirname, '..', 'resources', 'start');
-const webpackDir = project.dir.join('.webpack');
+const webpackDir = new FilePath(cfg.getStartDir(), {posix: true});
 const pkgJson = project.pkgJson;
 
+const msg = {
+	alreadyEjected: [
+		'',
+		error`
+		The {npm run start} configuration has already been ejected !
+		`,
+		`
+		No modifications have been done to any project file.
+		`,
+	],
+
+	ejectFinished: [
+		success`
+		The {npm run start} configuration has been successfully ejected.
+		`,
+	],
+	ejectWarning: [
+		'',
+		warn`
+		You have decided to {eject} the {npm run start} configuration.
+		`,
+		`
+		This will regenerate it for the last time, and leave everything in the
+		{.webpack} directory so that you can manually configure and tweak it
+		later.
+
+		Also, the {.webpack} directory will be removed from {.gitignore} if it
+		exists in the project folder. If the project does not have a
+		{.gitignore} file, don't forget to modify the correct {.gitignore} file
+		yourself.
+
+		It will also note that an {eject} was performed, inside {.npmbuildrc}
+		file, remove {webpack} configuration, out of that file too, and point
+		{npm run start} to the {.webpack} folder.
+
+		After the {eject} you can change the name of the {.webpack} folder
+		(updating the {.npmbuildrc} accordingly) if desired.
```

Comment:
Wordsmithing a bit, let me know if this makes sense.

```
		warn`
		You have decided to {eject} the {npm run start} configuration.
		`,
		`
		This will regenerate your webpack config for the last time, and leave it in the
		{.webpack} directory so that you can manually configure and tweak it.

		The {.webpack} directory will be removed from {.gitignore} if it
		exists in your project folder. If your project does not have a
		{.gitignore} file, don't forget to modify the correct {.gitignore} file
		yourself.

        Your {.npmbuildrc} file will also note that an {eject} was performed.
        Remove your {webpack} configuration from that file and point
		{npm run start} to the {.webpack} folder.

		After you {eject} you can change the name of the {.webpack} directory
		(update the {.npmbuildrc} accordingly) if desired.
```

Reply:
I tried using GH's suggestion syntax for a diff but it looked terrible

Reply:
Looks better :+1: . I only find that there's a comma missing here:

>  Your {.npmbuildrc} file will also note that an {eject} was performed. Remove your {webpack} configuration from that file and point {npm run start} to the {.webpack} folder.

It should say:

>  Your {.npmbuildrc} file will also note that an {eject} was performed, remove your {webpack} configuration from that file and point {npm run start} to the {.webpack} folder.

Because it's the process which removes the webpack configuration from the file. If we put a period instead of a comma some people may think that it's their duty to remove that config (imperative).

Sounds sensible?

Reply:
In fact, it is still confusing, isn't it? 

Maybe something like this:

> The process will also write (in your {.npmbuildrc} file) that an {eject} was performed, remove your {webpack} configuration, and configure {.webpack}  as your `npm run start` work directory.

Reply:
Feel free to wordsmith it a bit more to make it sound more English :sweat_smile:

Reply:
I've pushed a modification. Let me know how it looks...

Reply:
> there's a comma missing here

gonna be honest here for a moment... I'm not totally confident in my knowledge of when to use commas.


Just read the modification and it looks good to me.

Reply:
Good then :+1: . I'll trust your English and you trust my commas :sweat_smile: